mattermost-plugin/webapp/src/index.tsx

```diff
@@ -143,6 +151,50 @@ export default class Plugin {
             }, '', 'Boards')
             this.registry.registerCustomRoute('/', MainApp)
         }
+
+        const config = await octoClient.getClientConfig()
+        if (config?.telemetry) {
+            let rudderKey = TELEMETRY_RUDDER_KEY
+            let rudderUrl = TELEMETRY_RUDDER_DATAPLANE_URL
+
+            if (rudderKey.startsWith('placeholder') && rudderUrl.startsWith('placeholder')) {
+                rudderKey = process.env.RUDDER_KEY as string //eslint-disable-line no-process-env
+                rudderUrl = process.env.RUDDER_DATAPLANE_URL as string //eslint-disable-line no-process-env
+            }
+
+            if (rudderKey != null && rudderKey !== '') {
```

Comment:
Doesn't seem possible that `rudderKey` is `null`, as it is being set to `''` if there is no env variable in `webpack.config`
```suggestion
            if (rudderKey !== '') {
```

Reply:
done